### PR TITLE
Reintegrate New Alerts, fix whitespace issues

### PR DESCRIPTION
**Pull request by zdeyn**
Hopefully we can use this to wodge the features in - HELP! lol

Old Alerts:
- Hardcore_Notification_Frame
- Hardcore_Warning_Frame

New Alerts:
- Hardcore_Alert_Frame

Styles:
- logo (semi-broken, just a bad default)
- death
- hc_green
- hc_red
- spirithealer
- (do we need more?)

Alerts in use within AddOn:
- DONE: Hardcore:RecordReminder() -> hc_green
- DONE: Hardcore:Add(data) -> death
- DONE: Hardcore:PLAYER_UNGHOST -> spirithealer
- DONE: Hardcore:UNIT_SPELLCAST_START -> hc_red
- DONE: Bubble-hearth -> hc_red / hc_green (cancel)

### CODE REVIEW

File: Hardcore.lua [hunk 1]
```diff
@@ -370,78 +376,83 @@ function Hardcore:PLAYER_DEAD()
...
-	end)
+	Hardcore:ShowAlertFrame(ALERT_STYLES.spirithealer, message)
+
+    -- Screen notification
+    -- Hardcore_Notification_Text:SetText(message)
+    -- Hardcore_Notification_Frame:Show()
+    -- PlaySound(8959)
+    -- C_Timer.After(COMM_DELAY, function()
+    --     Hardcore_Notification_Frame:Hide()
+    -- end)
```
Comment (blenderer) [inline Hardcore.lua:408]: Can we remove this?

File: Hardcore.lua [hunk 2]
```diff
@@ -279,18 +277,20 @@ function Hardcore:UNIT_SPELLCAST_START(...)
...
+				Hardcore:Print("WARNING: Bubble-hearth Detected\nCancel or risk invalidation")
+
+				Hardcore:ShowAlertFrame(ALERT_STYLES.hc_red, "Bubble-hearth Detected\nCancel or risk invalidation")
+
+				-- Hardcore_Warning_Text:SetText("Started bubble hearthing \n Cancel or run will be invalidated.")
+				-- Hardcore_Warning_Frame:Show()
+				-- PlaySound(8959)
+				-- C_Timer.After(8, function()
+				--	 Hardcore_Warning_Frame:Hide()
+				-- end)
```
Comment (blenderer) [inline Hardcore.lua:290]: Can we remove this as well?

File: Hardcore.lua [hunk 3]
```diff
@@ -663,12 +686,14 @@ function Hardcore:Add(data)
...
-			Hardcore_Notification_Frame:Hide()
-		end)
+		Hardcore:ShowAlertFrame(ALERT_STYLES.death, messageString)
+
+        -- Hardcore_Notification_Text:SetText(messageString)
+        -- Hardcore_Notification_Frame:Show()
+        -- PlaySound(8959)
+        -- C_Timer.After(COMM_DELAY, function()
+        --     Hardcore_Notification_Frame:Hide()
+        -- end)
```
Comment (blenderer) [inline Hardcore.lua:696]: Remove?

File: Hardcore.lua [hunk 4]
```diff
@@ -1048,15 +1078,21 @@ function Hardcore_Deathlist_ScrollBar_Update()
...
-		Hardcore_Notification_Frame:Hide()
-	end)
+	Hardcore:ShowAlertFrame(ALERT_STYLES.hc_green, "Hardcore Enabled\n START RECORDING")
+
+    -- Hardcore_Notification_Text:SetText()
+    -- Hardcore_Notification_Frame:Show()
+    -- PlaySound(8959)
+    -- C_Timer.After(10, function()
+    --     Hardcore_Notification_Frame:Hide()
+    -- end)
```
Comment (blenderer) [inline Hardcore.lua:1095]: Remove?

**Review by blenderer — COMMENTED** [review]
Looks good, can we remove the commented-out code?

File: Hardcore.lua (hunk 1 above)
Comment (zdeyn) [reply Hardcore.lua]: Yep, it's pure confidence-code - I'd prefer to remove after testing, but w/e works for you

**Comment by zdeyn** [comment]
(from PR #24 )
The alerts I've replaced are still in-code, commented out so comparisons can be made during testing - easy remove once happy.

Old Alerts:
- Hardcore_Notification_Frame
- Hardcore_Warning_Frame

New Alerts:
- Hardcore_Alert_Frame

Styles:
- logo (semi-broken, just a bad default)
- death
- hc_green
- hc_red
- spirithealer
- (do we need more?)

Alerts in use within AddOn:
- DONE: Hardcore:RecordReminder() -> hc_green
- DONE: Hardcore:Add(data) -> death
- DONE: Hardcore:PLAYER_UNGHOST -> spirithealer
- DONE: Hardcore:UNIT_SPELLCAST_START -> hc_red

**Review by blenderer — COMMENTED** [review]
Looks good, going to pull down and test locally

**Comment by aaronma37** [comment]
I think we should all try to use the same auto linter or none, otherwise I think a lot of PRs will have to be rebased

**Comment by zdeyn** [comment]
> I think we should all try to use the same auto linter or none, otherwise I think a lot of PRs will have to be rebased

Completely agree, we'll figure it after we ship 0.5.0

**Pull request merged**